source/basic.tex

```diff
@@ -1396,583 +1355,620 @@
-\tcode{X}\iref{class.static.data} (after the \grammarterm{qualified-id}
-of the static member) is looked up as if the name was used in a member
-function of \tcode{X}.
+A \grammarterm{using-directive} is
+\term{active} in a scope $S$ at a program point $P$
+if it precedes $P$ and inhabits either $S$ or
+the scope of a namespace nominated by a \grammarterm{using-directive}
+that is active in $S$ at $P$.
 \begin{note}
-\ref{class.static.data} further
-describes the restrictions on the use of names in the definition of a
-\tcode{static} data member.
+A \grammarterm{using-directive} is exported if and only if
+it appears in a header unit.
 \end{note}
 
 \pnum
-If a variable member of a namespace is defined outside of the scope of
-its namespace then any name that appears in the definition of the
-member (after the \grammarterm{declarator-id}) is looked up as if the
-definition of the member occurred in its namespace.
+An \term{unqualified search} in a scope $S$ from a program point $P$
+includes the results of searches from $P$ in
+\begin{itemize}
+\item
+$S$, and
+\item
+for any scope $U$ that contains $P$ and is or is contained by $S$,
+each namespace contained by $S$ that is nominated by
+a \grammarterm{using-directive} that is active in $U$ at $P$.
+\end{itemize}
+If no declarations are found,
+the results of the unqualified search are
+the results of an unqualified search in the parent scope of $S$, if any,
+from $P$.
+\begin{note}
+When a class scope is searched,
+the scopes of its base classes are also searched\iref{class.member.lookup}.
+If it inherits from a single base,
+it is as if the scope of the base immediately contains
+the scope of the derived class.
+Template parameter scopes
+that are associated with one scope in the chain of parents
+are also considered\iref{temp.local}.
+\end{note}
+
+\pnum
+\defnx{Unqualified name lookup}{lookup!unqualified name}
+from a program point performs an unqualified search in its immediate scope.
+
+\pnum
+An \defnadj{unqualified}{name} is a name
+that does not follow a \grammarterm{nested-name-specifier} or
+the \tcode{.} or \tcode{->} in a class member access expression\iref{expr.ref},
+possibly after a \keyword{template} keyword or \tcode{\~}.
+Unless otherwise specified,
+such a name undergoes unqualified name lookup from the point where it appears.
+
+\pnum
+An unqualified name that is a component name\iref{expr.prim.id.unqual} of
+a \grammarterm{type-specifier} or \grammarterm{ptr-operator} of
+a \grammarterm{conversion-type-id} is looked up in the same fashion
+as the \grammarterm{conversion-function-id} in which it appears.
+If that lookup finds nothing, it undergoes unqualified name lookup;
+in each case, only names
+that denote types or templates whose specializations are types are considered.
+
+\pnum
+In a friend declaration \grammarterm{declarator}
+whose \grammarterm{declarator-id} is a \grammarterm{qualified-id}
+whose lookup context is\iref{basic.lookup.qual} a class or namespace $S$,
```

Comment:
This reproduces a mistake in cross-reference placement from the paper (sorry):
```suggestion
whose lookup context\iref{basic.lookup.qual} is a class or namespace $S$,
```

Reply:
Ah, sorry, I missed that, will fix after merges!

Reply:
Fixed by 2432acacbce559ac6cd3e2878ff68e3b917484ec, thanks!